rmw_opensplice_cpp/src/rmw_service_server_is_available.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <ccpp_dds_dcps.h>
+#include <dds_dcps.h>
+
+#include "rmw/allocators.h"
+#include "rmw/error_handling.h"
+#include "rmw/impl/cpp/macros.hpp"
+#include "rmw/rmw.h"
+
+#include "identifier.hpp"
+#include "types.hpp"
+
+// The extern "C" here enforces that overloading is not used.
+extern "C"
```

Comment:
do you need to wrap this with `#if __cplusplus` here or is that only needed in header files?